agent/mibgroup/if-mib/data_access/interface.c

```diff
@@ -782,6 +790,40 @@ netsnmp_access_interface_entry_overrides(netsnmp_interface_entry *entry)
     }
 }
 
+/**---------------------------------------------------------------------*/
+/*
+ * include_ifmib_iface_prefix config token
+ *
+ *     If there is an iface name match, we return 1 for True
+ *     otherwise return 0 for False.
+ *     If there are no ifaces defined at all, return 1 so that the
+ *     default behavior is to include all ifaces (include everything).
+ *     (Note: including at least one iface name means we will only include
+ *     those iface names and exclude all others.)
+ *
+ */
+int netsnmp_access_interface_include(const char * name)
+{
+    netsnmp_include_if_list * if_ptr;
+
+    netsnmp_assert(1 == _access_interface_init);
+    if (NULL == name)
+        return 1;
+
+    if (include_list == NULL)
+        /*
+         * If include_ifmib_iface_prefix was not configured, we should include
+         * all interfaces (which is the default).
+         */
+        return 1;
+
+    for (if_ptr = include_list; if_ptr; if_ptr = if_ptr->next)
+        if (memcmp(name, if_ptr->name, strlen(if_ptr->name)) == 0)
```

Comment:
Can this memcmp() call be changed into a strncmp() call? I'm concerned that memcmp() could read past the end of 'name' and that that could trigger a Valgrind complaint.

Reply:
Yes, I can change it.